Source/mpqapi.cpp

```diff
@@ -31,28 +31,20 @@ BOOL mpqapi_set_hidden(const char *pszArchive, BOOL hidden)
 
 void mpqapi_store_creation_time(const char *pszArchive, int dwChar)
 {
-	int v2;                                // esi
-	const char *v3;                        // ebx
-	HANDLE v4;                             // eax
-	int v5;                                // esi
-	struct _WIN32_FIND_DATAA FindFileData; // [esp+8h] [ebp-1E0h]
-	char dst[160];                         // [esp+148h] [ebp-A0h]
-
-	v2 = dwChar;
-	v3 = pszArchive;
+	HANDLE handle;
+	struct _WIN32_FIND_DATAA FindFileData;
+	char dst[160];
+
 	if (gbMaxPlayers != 1) {
 		mpqapi_reg_load_modification_time(dst, 160);
-		v4 = FindFirstFile(v3, &FindFileData);
-		if (v4 != INVALID_HANDLE_VALUE) {
-			FindClose(v4);
-			v5 = 16 * v2;
-			*(_DWORD *)&dst[v5] = FindFileData.ftCreationTime.dwLowDateTime;
-			*(_DWORD *)&dst[v5 + 4] = FindFileData.ftCreationTime.dwHighDateTime;
+		handle = FindFirstFile(pszArchive, &FindFileData);
+		if (handle != INVALID_HANDLE_VALUE) {
+			FindClose(handle);
+			*((FILETIME *)(dst) + dwChar * 2) = FindFileData.ftCreationTime;

```

Comment:
We are storing `ftCreationTime` which is a DWORD into the regedit, not the pointer of `FindFileData`, thus I've updated the cast.

```suggestion
			*((DWORD *)(dst) + dwChar * 2) = FindFileData.ftCreationTime;
```

Reply:
`DWORD` is 4 bytes, ftCreationTime is a `struct` made of two `DWORD` fields. For bin-exactness we need 8-byte store and not two 4-byte stores at least to my understanding. I could've casted both sides to `__int64*` but I decided that it looked pretty silly.

Reply:
Oh, you are right indeed. I mistook ftCreationTime as a DWORD. It is a [FILETIME](https://docs.microsoft.com/en-us/windows/desktop/api/minwinbase/ns-minwinbase-filetime) struct, which is indeed 8 bytes in size, as you indicated.